tools/wptserve/wptserve/server.py

```diff
@@ -29,6 +29,15 @@
 from .utils import HTTPException
 from .constants import h2_headers
 
+# We need to stress test browsers to send/receive lots of headers, but Python
+# stdlib has an arbitrary limit of 100 headers. Hitting the limit would produce
+# an exception which is silently caught in Python 2 but leads to HTTP 431 in
+# Python 3, so we have to moneky patch this limit.
+# See also https://github.com/web-platform-tests/wpt/pull/24451 .
```

Comment:
I think this could be reworded sightly (and fix a typo)

```suggestion
# We need to stress test that browsers are able to send/receive many headers
# (there is no specified limit), but the Python stdlib has an arbitrary limit of 100
# headers. Hitting the limit would produce an exception that is silently caught
# in Python 2 but leads to HTTP 431 in Python 3, so we monkey patch it higher.
# See also https://github.com/web-platform-tests/wpt/pull/24451 .
```

Also, where was the exception caught? Did we never support >100 headers but just never noticed because the exception was swallowed?

Reply:
Deep within stdlib, which was arguably a bug and was "fixed" in Python 3.6 which led to this breakage. I'll add a link to the change.

Reply:
Also to your other question: I did not chase too deep, but Py2 passes my newly added test without the need for the monkey patch, so it must support 100+ headers.